linchpin_api/v1/api.py

```diff
@@ -0,0 +1,214 @@
+import os
+import inspect
+import ansible
+import pprint
+import linchpin_api
+from tabulate import tabulate
+from ansible import utils
+import jsonschema as jsch
+from collections import namedtuple
+from ansible import utils
+from ansible.parsing.dataloader import DataLoader
+from ansible.vars import VariableManager
+from ansible.inventory import Inventory
+from ansible.executor.playbook_executor import PlaybookExecutor
+from ansible.plugins.callback import CallbackBase
+from callbacks import PlaybookCallback
+from invoke_playbooks import invoke_linchpin
+from utils import get_file, list_files, parse_yaml
+from github import GitHub
+
+
+class LinchpinAPI:
+
+    def __init__(self):
+        self.PLAYBOOKS = {
+                           "PROVISION": "site.yml",
+                           "TEARDOWN": "site.yml",
+                           "SCHEMA_CHECK": "schema_check.yml",
+                           "INVGEN": "invgen.yml",
+                         }
+        base_path = os.path.dirname(__file__).split("/")
+        base_path = base_path[0:-2]
+        self.base_path = "/".join(base_path)
+        self.excludes = set(["topology_upstream",
+                             "layout_upstream",
+                             "post_actions"])
+
+    def get_config_path(self):
+        try:
+            config_files = [
+                            "./linchpin_config.yml",
+                            "~/.linchpin_config.yml",
+                            self.base_path+"/linchpin_config.yml",
+                            "/etc/linchpin_config.yml"]
+            for c_file in config_files:
+                if os.path.isfile(c_file):
+                    return c_file
+        except Exception as e:
+            print(e)
+
+    def get_config(self):
+        config_path = self.get_config_path()
+        config = parse_yaml(config_path)
+        return config
+
+    def get_evars(self, lpf):
+        """ creates a group of extra vars on basis on linchpin file dict """
+        e_vars = []
+        for group in lpf:
+            topology = lpf[group].get("topology")
+            layout = lpf[group].get("layout")
+            e_var_grp = {}
+            e_var_grp["topology"] = search_path(topology, os.getcwd())
+            e_var_grp["layout"] = search_path(layout, os.getcwd())
+            if None in e_var_grp.values():
+                display("ERROR:003")
```

Comment:
Additionally, I'm not a fan of the ERROR:003 concept here. It makes me look up error messages to figure out where the bug is happening. I'd rather have very specific errors for each situation. If a topology is missing, say it's missing. If it's a layout that doesn't parse properly, let's give a good error message there, with as much detail as possible to help troubleshoot.

Users don't necessarily know what all of that means, but they can read english and determine why something doesn't work. Let's give them the benefit of the doubt and assume they are smart enough to read a traceback and a clean error message.